src/System.Transactions.Local/src/System/Transactions/TransactionState.cs

```diff
@@ -263,7 +263,7 @@ internal virtual void AddOutcomeRegistrant(InternalTransaction tx, TransactionCo
 
         internal virtual void GetObjectData(InternalTransaction tx, SerializationInfo serializationInfo, StreamingContext context)
         {
-            throw TransactionException.CreateTransactionStateException(tx._innerException, tx.DistributedTxId);
+            throw new PlatformNotSupportedException();
```

Comment:
Please don't change this. If this "base" method executes, it is an invalid state error. That won't change even when we end up supporting serialization.